src/kirby/components/modal/modal.component.ts

```diff
@@ -0,0 +1,146 @@
+import {  Component,
+          OnInit,
+          OnDestroy,
+          Input,
+          Output,
+          EventEmitter,
+          ViewChild,
+          ElementRef,
+          AfterViewChecked,
+        } from '@angular/core';
+
+@Component({
+  selector: 'kirby-modal',
+  templateUrl: './modal.component.html',
+  styleUrls: ['./modal.component.scss']
+})
+export class ModalComponent implements OnInit, OnDestroy, AfterViewChecked {
+  // State of the modal - called from another component, e.g. a button
+  private _isOpen = false;
+
+  @Input() size?: 'small' | 'medium' | 'large';
+  // Theme option - e.g. dark, light or whatever theme there is
+  @Input() theme?: string;
+  modalCssClasses: {};
+  // Output closing state
+  @Output() closingModal = new EventEmitter();
+  // Pointer to modal DOM element
+  private activeModal: ElementRef;
+  // isFocus flag to prevent ngAfterViewChecked from running multiple times
+  private isFocus: boolean;
+  // Remember which element had focus before opening the modal
+  private oldFocus: HTMLElement;
+
+  @ViewChild('activeModal') set content(content: ElementRef) {
+    this.activeModal = content;
+  }
+
+  ngOnInit() {
+    const { theme } = this;
+    let { size } = this;
+
+    if (size === void 0) {
+      size = 'medium';
+    }
+    this.modalCssClasses = {
+      ['modal--' + size]: size,
+      ['theme--' + theme] : theme,
+    };
+    this.keydownHandler = this.keydownHandler.bind(this);
+    this.keyupHandler = this.keyupHandler.bind(this);
+  }
+
+  ngOnDestroy() {
+    document.removeEventListener('keydown', this.keydownHandler, false);
```

Comment:
You shouldn't access document directly. Use [@HostListener](https://angular.io/api/core/HostListener) instead.